### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by vanelnw**
1. Remove current_user method in ApplicationController, devise will provide us with one.
2. Install and setup devise

- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.
3. modifie the devise view

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (mukhammadrizooff) [inline app/models/user.rb]: - [ ] [REQUIRED] Good jo so far! The application asks for `confirmation` of email. You should enable`:confirmable` in the user model (`user.rb` file) 🙏🏼 Your code looks like something similar to what is below:

```ruby
	devise :database_authenticatable, :registerable,
				:recoverable, :rememberable, :validatable, :confirmable
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @vanelnw, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct `GitFlow` ✔️ 
- Good implementations ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
+</div>
```
Comment (Whoistolu) [inline app/views/devise/registrations/new.html.erb:34]: - [ ] [OPTIONAL] Great job so far @vanelnw 👍🏽 , However, as stated by the project requirements, you need to modify your design to match the wireframe provided. Please also take note that I could not signup despite the fact that I provided all the credentials needed!
- It is very important that you refactor your code to achieve this.

Your Implementation   |  Figma Design
:-------------------------:|:-------------------------:
Screenshot 2023-03-09 at 19 53 09 | Screenshot 2023-03-09 at 19 51 38

**Review by Whoistolu — APPROVED** [review]
### Hi @vanelnw 

Great job implementing the recommended changes by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

- [ ] [OPTIONAL] I will recommend that you remove the second author in the `Author` section of the README since you are the only one that worked on this milestone.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 2 above)
Comment (vanelnw) [reply app/views/devise/registrations/new.html.erb:34]: hi @Whoistolu 
Thank you for mentioning all these amelioration, i will work on it

**Pull request merged**